skills_ml/algorithms/occupation_classifiers/train.py

```diff
@@ -0,0 +1,131 @@
+                'sklearn.ensemble.ExtraTreesClassifier': {
+                    'n_estimators': [100, 500],
+                    'criterion': ['gini', 'entropy'],
+                    'max_depth': [1, 5, 10, 20, 50],
+                    'max_features': ['sqrt', 'log2'],
+                    'min_samples_split': [2, 5, 10]
+                    },
+                'sklearn.ensemble.RandomForestClassifier': {
+                    'n_estimators': [100, 500],
+                    'criterion': ['gini', 'entropy'],
+                    'max_depth': [1, 5, 10, 20, 50],
+                    'max_features': ['sqrt', 'log2'],
+                    'min_samples_split': [2, 5, 10]
+                    }
+                }
+
+    def train(self):
+        """Fit a model to a training set. Works on any modeling class that
+        is vailable in this package's environment and implements .fit
+        """
+        X = self.matrix['X']
+        y = self.matrix['y']
+        for score in self.scores:
+            for class_path, parameter_config in self.grid_config.items():
+                module_name, class_name = class_path.rsplit(".", 1)
+                module = importlib.import_module(module_name)
+                cls = getattr(module, class_name)
+                kf = StratifiedKFold(n_splits=self.k_folds, random_state=self.random_state_for_split)
+
+                cls_cv = GridSearchCV(cls(), parameter_config, cv=kf, scoring=score, n_jobs=self.n_jobs)
+                cls_cv.fit(X, y)
+                self.cls_cv_result[score] = {class_name: cls_cv.cv_results_}
+                self.best_classifiers[score] = {class_name: cls_cv}
+
+    def save_result(self):
+        return NotImplementedError
+
+
+class SocEncoder(LabelEncoder):
+    def __init__(self, label_list):
+        self.fit(label_list)
+
+
+def create_training_set(job_postings_generator, embedding_model=None, target_variable="major_group",
+                        document_schema_fields=['description','experienceRequirements', 'qualifications', 'skills']):
+    """Create training set for occupation classifier from job postings generator and embedding model
+
+    Args:
+        job_postings_generator (generator): job posting collection
+        emebdding_model (skills_ml.algorithm.embedding.models): embedding model
+        document_schema_fields (list): fields to be included in the training data
+
+    Returns:
+        (dict) a dictionary of training data(X), label(y), the embedding_model, target_variable and soc_encoder
+    """
+    X = []
+    y = []
+
+    if target_variable == "major_group":
+        se = SocEncoder(list(majorgroupname.keys()))
+        label_transformer = lambda soc_code: se.transform([soc_code[:2]])
+    elif target_variable == "full_soc":
+        se = SocEncoder(get_all_soc())
+        label_transformer = lambda soc_code: se.transform([soc_code])
+
+    for job in job_postings_generator:
+        if job['onet_soc_code'][:2] != '99' and job['onet_soc_code'] != '' :
```

Comment:
Maybe keep track of how many got skipped, and log it at the end so we can see what percentage of the corpus got dropped on the floor before training.